data.md

```diff
@@ -491,51 +491,42 @@ Byte Map
 Wasm memories are byte arrays, sized in pages of 65536 bytes, initialized to be all zero bytes.
 To avoid storing many zeros in what may be sparse memory, we implement memory as maps, and store only non-zero bytes.
 The maps store integers, but maintains the invariant that each stored integer is between 1 and 255, inclusive.
-`BM [ N := I ]` writes the integer `I` to memory as bytes (little-endian), starting at index `N`.
+`#setRange(BM, START, VAL, WIDTH)` writes the integer `I` to memory as bytes (little-endian), starting at index `N`.
 
 ```k
-    syntax Map ::= Map "[" Int ":=" Int "]" [function]
- // --------------------------------------------------
-    rule BM [ IDX := 0  ] => BM
-    rule BM [ IDX := VAL] => BM [IDX <- VAL modInt 256 ] [ IDX +Int 1 := VAL /Int 256 ]
-      requires VAL modInt 256 =/=Int 0
-       andBool VAL >Int 0
-    // Don't store 0 bytes.
-    rule BM [ IDX := VAL] => BM [IDX <- undef          ] [ IDX +Int 1 := VAL /Int 256 ]
-      requires VAL modInt 256 ==Int 0
-       andBool VAL >Int 0
+    syntax Map ::= #setRange(Map, Int, Int, Int) [function]
+ // -------------------------------------------------------
+    rule #setRange(BM, IDX,   _, N) => BM
+      requires N  ==Int 0
+    rule #setRange(BM, IDX, VAL, N) => #setRange(#set(BM, IDX, VAL modInt 256), IDX +Int 1, VAL /Int 256, N -Int 1)
+      requires N =/=Int 0
```

Comment:
Let's use `N >Int 0` and `notBool (N >Int 0)` for side conditions, to (1) ensure completeness and (2) ensure termination.

Reply:
Done